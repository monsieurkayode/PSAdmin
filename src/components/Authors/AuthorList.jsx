import React from 'react';
import { arrayOf, shape, func } from 'prop-types';
import AuthorListRow from './AuthorListRow';

const AuthorList = ({ authors, deleteAuthor }) => (
  <table className="table">
    <thead>
      <tr>
        <th>&nbsp;</th>
        <th>ID</th>
        <th>Author Name</th>
        <th>&nbsp;</th>
      </tr>
    </thead>
    <tbody>
      {authors.map(author => (
        <AuthorListRow
          key={author.id}
          {...author}
          deleteAuthor={deleteAuthor}
        />
      ))}
    </tbody>
  </table>
);

AuthorList.propTypes = {
  authors: arrayOf(shape({})).isRequired,
  deleteAuthor: func.isRequired,
};

export default AuthorList;
