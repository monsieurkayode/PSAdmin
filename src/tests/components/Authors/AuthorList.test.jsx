import React from 'react';
import { shallow } from 'enzyme';
import AuthorList from '../../../components/Authors/AuthorList';
import mockAuthors from '../../__mocks__/mockAuthors';

const setup = () => {
  const props = {
    authors: mockAuthors,
    deleteAuthor: jest.fn()
  };

  return shallow(<AuthorList {...props} />);
};

describe('AuthorList Component', () => {
  const wrapper = setup();

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a table with a table row having four table header cells', () => {
    expect(wrapper.find('table').length).toBe(1);
    expect(wrapper.find('th').length).toBe(4);
  });

  it('has table headings ID, Author Name', () => {
    expect(wrapper.find('tr').childAt(1).text()).toEqual('ID');
    expect(wrapper.find('tr').childAt(2).text()).toEqual('Author Name');
  });

  it('renders 2 AuthorListRow components', () => {
    expect(wrapper.find('AuthorListRow').length).toEqual(2);
  });
});
