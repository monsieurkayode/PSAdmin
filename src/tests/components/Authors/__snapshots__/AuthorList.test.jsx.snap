// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorList Component renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
         
      </th>
      <th>
        ID
      </th>
      <th>
        Author Name
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody>
    <AuthorListRow
      deleteAuthor={[MockFunction]}
      firstName="Cory"
      id="cory-house"
      key="cory-house"
      lastName="House"
    />
    <AuthorListRow
      deleteAuthor={[MockFunction]}
      firstName="Scott"
      id="scott-allen"
      key="scott-allen"
      lastName="Allen"
    />
  </tbody>
</table>
`;
